Source/WTF/wtf/OptionSet.h

```diff
@@ -242,10 +184,22 @@ template<typename E> class OptionSet {
     StorageType m_storage { 0 };
 };
 
+namespace IsValidOptionSetHelper {
+template<typename T, typename E> struct OptionSetValueChecker;
+template<typename T, typename E, E e, E... es>
+struct OptionSetValueChecker<T, EnumValues<E, e, es...>> {
+    static constexpr T allValidBits() { return static_cast<T>(e) | OptionSetValueChecker<T, EnumValues<E, es...>>::allValidBits(); }
+};
+template<typename T, typename E>
+struct OptionSetValueChecker<T, EnumValues<E>> {
+    static constexpr T allValidBits() { return 0; }
+};
+}
+
 template<typename E>
 WARN_UNUSED_RETURN constexpr bool isValidOptionSet(OptionSet<E> optionSet)
 {
-    auto allValidBitsValue = OptionSetValueChecker<std::make_unsigned_t<std::underlying_type_t<E>>, typename EnumTraits<E>::values>::allValidBits();
+    auto allValidBitsValue = IsValidOptionSetHelper::OptionSetValueChecker<std::make_unsigned_t<std::underlying_type_t<E>>, typename EnumTraits<E>::values>::allValidBits();
```

Comment:
Does this EnumTraits use need removed?

Reply:
It does, but not until all deserialized OptionSet enums are migrated to generated serialization.  I added a FIXME comment indicating such.